src/controllers/order_controller.go

```diff
@@ -0,0 +1,136 @@
+package controllers
+
+import (
+	"github.com/ELITE-Kinoticketsystem/Backend-KTS/src/.gen/KinoTicketSystem/model"
+	kts_errors "github.com/ELITE-Kinoticketsystem/Backend-KTS/src/errors"
+	"github.com/ELITE-Kinoticketsystem/Backend-KTS/src/models"
+	"github.com/ELITE-Kinoticketsystem/Backend-KTS/src/repositories"
+	"github.com/ELITE-Kinoticketsystem/Backend-KTS/src/utils"
+	"github.com/google/uuid"
+)
+
+type OrderControllerI interface {
+	CreateOrder(CreateOrderDTO models.CreateOrderDTO, eventId *uuid.UUID, userId *uuid.UUID, isReservation bool) (*uuid.UUID, *models.KTSError)
+	GetOrderById(orderId *uuid.UUID, userId *uuid.UUID) (*models.GetOrderDTO, *models.KTSError)
+	GetOrders(userId *uuid.UUID) (*[]models.GetOrderDTO, *models.KTSError)
+}
+
+type OrderController struct {
+	OrderRepo         repositories.OrderRepoI
+	EventSeatRepo     repositories.EventSeatRepoI
+	PriceCategoryRepo repositories.PriceCategoryRepositoryI
+	TicketRepo        repositories.TicketRepoI
+}
+
+func (oc *OrderController) CreateOrder(CreateOrderDTO models.CreateOrderDTO, eventId *uuid.UUID, userId *uuid.UUID, isReservation bool) (*uuid.UUID, *models.KTSError) {
+	if isReservation && CreateOrderDTO.PaymentMethodID != nil {
+		return nil, kts_errors.KTS_BAD_REQUEST
+	}
+
+	slectedSeats, kts_err := oc.EventSeatRepo.GetSelectedSeats(eventId, userId)
+	if kts_err != nil {
+		return nil, kts_err
+	}
+
+	priceCategories, kts_err := oc.PriceCategoryRepo.GetPriceCategories()
+	if kts_err != nil {
+		return nil, kts_err
+	}
+
+	adultPriceCategory := getPriceCategoryByName(*priceCategories, utils.ADULT)
+	orderId := utils.NewUUID()
+
+	tickets, totalPrice := createTicketsAndCalculateTotalPrice(slectedSeats, CreateOrderDTO, priceCategories, adultPriceCategory, orderId)
+
+	order := model.Orders{
+		ID:              orderId,
+		UserID:          userId,
+		PaymentMethodID: CreateOrderDTO.PaymentMethodID,
+		IsPaid:          !isReservation,
```

Comment:
This is a bit complicated because you need to know that this is going to be turned around. Rather, change if statement above and let the user switch it in the handler if that is necessary.